meilisearch-http/src/routes/indexes/documents.rs

```diff
@@ -274,3 +274,23 @@ pub async fn clear_all_documents(
     debug!("returns: {:?}", task);
     Ok(HttpResponse::Accepted().json(task))
 }
+
+/// Returns the Content-Type header value without the charset=utf-8 or anything,
+/// only the value that is before the first semicolon (;).
+///
+/// Here is the composition of an HTTP Content-Type header
+/// <https://datatracker.ietf.org/doc/html/rfc7231#section-3.1.1.1>.
+fn extract_content_type(headers: &HeaderMap) -> Option<String> {
+    match headers.get(CONTENT_TYPE) {
+        Some(value) => match value.to_str() {
+            Ok(content) => Some(
+                content
+                    .split_once(";")
+                    .map_or(content, |(ct, _)| ct.trim())
+                    .to_string(),
+            ),
+            Err(_) => Some(format!("{}", value.as_bytes().as_bstr())), // convenient bytes display
+        },
+        None => None,
```

Comment:
I would prefer that we use [this method](https://docs.rs/actix-web/3.3.2/actix_web/trait.HttpMessage.html#method.mime_type) that is implemented on `HttpRequest`, and leverage the Mine library, instead of doing this ourselves

Reply:
Wow, didn't see this, that's awesome. The only difference that I see with my function is that it will not try to return a malformed content-type, one that is not utf-8 encoded for example, and we will not be able to return it in the error messages. But, not sure that's very important. Changing my PR to use this method, thank you!